### PR TITLE
10-Enable Redis Ready Pattern to be set as part of Redis Exec Provider.

### PR DESCRIPTION
**Pull request by xhoong**
Issue #10 PR, please review.

The idea is to extend RedisExecProvider to hold the executable name and it's server ready pattern.

### CODE REVIEW

File: src/main/java/redis/embedded/AbstractRedisInstance.java [hunk 1]
```diff
@@ -127,6 +127,8 @@ private void cancelStartupTimeoutCheck(ScheduledExecutorService executor) {
 
     protected abstract Pattern redisReadyPattern();
 
+    public abstract void setRedisReadyPat(Pattern redisReadyPat);
```
Comment (fmonniot) [inline src/main/java/redis/embedded/AbstractRedisInstance.java]: I'd like to have this set via the `Builder` pattern, and not be able to mutate this class. Could you remove this method, and set the `Pattern` via the `Builder`s instead ?

File: src/main/java/redis/embedded/RedisSentinel.java [hunk 2]
```diff
@@ -22,14 +22,20 @@ public RedisSentinel(List<String> args, int port) {
         this.args = new ArrayList<>(args);
     }
 
+    @Override
+    public void setRedisReadyPat(Pattern redisReadyPat) {
```
Comment (fmonniot) [inline src/main/java/redis/embedded/RedisSentinel.java]: same as before, it should be set in the `Builder` and passed via the class constructor.

File: src/main/java/redis/embedded/RedisExecProvider.java [hunk 3]
```diff
@@ -2,18 +2,22 @@
...
 import java.io.File;
 import java.io.IOException;
 import java.util.Map;
+import java.util.regex.Pattern;
 
 public class RedisExecProvider {
 
-    private final Map<OsArchitecture, String> executables = Maps.newHashMap();
+    private final Map<OsArchitecture, RedisExecutor> executables = Maps.newHashMap();
+    public final static Pattern REDIS_READY_PATTERN = Pattern.compile(
```
Comment (fmonniot) [inline src/main/java/redis/embedded/RedisExecProvider.java]: More of nitpick than anything, but could you name it `DEFAULT_REDIS_READY_PATTERN` ?
That way we really convey that it's overridable.

File: src/main/java/redis/embedded/util/RedisExecutor.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
+package redis.embedded.util;
+
+import java.util.regex.Pattern;
+
+public class RedisExecutor {
```
Comment (fmonniot) [inline src/main/java/redis/embedded/util/RedisExecutor.java]: The `Executor` part of the name make me think of `java.util.concurrent.Executor`, which lead to think it control how the process is run.
Could it be renamed to `RedisExecutable` (or something like that) ?

**Review by fmonniot — CHANGES_REQUESTED** [review]
Thanks for the PR !
I found some changes to do, but nothing too dramatic.
Good work on that one 👍

**Comment by xhoong** [comment]
Any chances a publish a release binary soon? Thanks

**Comment by xhoong** [comment]
Test runs ok on my side, can the check reinitiated? thanks.

**Review by fmonniot — APPROVED** [review]

**Comment by fmonniot** [comment]
I'll try to make a release this week, thanks for the contribution !

**Pull request merged**